pkg/docker/config/config.go

```diff
@@ -216,12 +216,25 @@ func getAuthFilePaths(sys *types.SystemContext, homeDir string) []authPath {
 // file or .docker/config.json, including support for OAuth2 and IdentityToken.
 // If an entry is not found, an empty struct is returned.
 func GetCredentials(sys *types.SystemContext, registry string) (types.DockerAuthConfig, error) {
-	return getCredentialsWithHomeDir(sys, registry, homedir.Get())
+	return getCredentialsWithHomeDir(sys, registry, "", homedir.Get())
+}
+
+// GetCredentialsWithRefPath returns the registry credentials stored in the
+// registry-specific credential helpers or in the default global credentials
+// helpers with falling back to using either auth.json
+// file or .docker/config.json, including support for OAuth2 and IdentityToken.
+// Compared to GetCredentials, this function accepts a refPath (containing the
+// full reference path including the image name) together with the registry.
+// This refPath will be taken into account when looking up the credentials in
```

Comment:
I think we need to elaborate on what "taken into account" exactly means.

Reply:
I agree and updated the docs for this function.